### PR TITLE
fix: update copy on the home page; small a11y fixes

### PR DESCRIPTION
**Pull request by meganesu**
This PR updates the copy on the home page, to make the call to action clearer (and include a Deploy to Cloud button).

It also adds some small accessibility fixes:

* Adding a visually hidden h1 to the search page.
* Give unique names to the `nav` elements for the footer and pagination.

https://app.clubhouse.io/gatsbyjs/story/30147/hero-h1-and-call-to-action-are-not-clear

This PR also updates the Deploy to Cloud link to include a UTM code:

https://app.clubhouse.io/gatsbyjs/story/30273/add-utm-to-the-deploy-to-gatsby-cta-link

### CODE REVIEW

File: src/pages/search-page.module.css [hunk 1]
```diff
@@ -1,3 +1,14 @@
+.visually-hidden {
+  border: 0;
+  clip: rect(0, 0, 0, 0);
+  height: 1px;
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  width: 1px;
+}
```
Comment (wardpeet) [inline src/pages/search-page.module.css:10]: I would move this to `a11y.module.css` and put it in some kind of styles/utils directory as you'll probably want to use it more than once. For now this is okay

**Review by wardpeet — APPROVED** [review]
LGTM! :)

File: src/pages/index.jsx [hunk 2]
```diff
@@ -16,19 +22,27 @@ export const query = graphql`
...
+        <p className={callOut}>
+          It's a proof-of-concept in a box, with 10k products and 30k variants
+          to help you get to proof-of-concept{" "}
+          <strong>as soon as right now</strong>.
+        </p>
+        <p className={callToAction}>
+          Hook it up to your own Shopify store data and start customizing in
+          minutes by deploying it to Gatsby Cloud for free. Grab your Shopify
+          store credentials and
+          <a href="https://www.gatsbyjs.com/dashboard/deploynow?url=https://github.com/gatsbyjs/gatsby-starter-shopify">
```
[the remark this answers is not in the document]
Comment (jxnblk) [reply src/pages/index.jsx:36]: We can probably tackle [ch30273] in this same PR if we also update the link in the footer

**Review by jxnblk — APPROVED** [review]
This looks good to me! If we can update the UTM parameters quickly, that's fine, but don't let that be a blocker

File: src/pages/search.jsx [hunk 3]
```diff
@@ -175,6 +176,7 @@ function SearchPage({
 
   return (
     <Layout>
+      <h1 className={visuallyHidden}>Search Results</h1>
```
[the remark this answers is not in the document]
Comment (fk) [reply src/pages/search.jsx:179]: yep!

**Comment by fk** [comment]
(on the run but — yay a11y fixes! <3 🙏)

File: src/pages/search.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (meganesu) [reply src/pages/search.jsx:179]: Meaning that this heading shouldn't be visually hidden? I added in this heading just for screen readers (there wasn't any h1 before, which was causing an error when I checked it with the axe dev tools).

There's a different `p` tag that has "Search results for X" lower down on the page (~ line 257). That one is visible to all users.

**Pull request merged**